go.mod

```diff
@@ -235,4 +235,4 @@ require (
 	sigs.k8s.io/yaml v1.2.0 // indirect
 )
 
-go 1.21
+go 1.21.3
```

Comment:
Not a reviewer but just curious, was pinning the minor version here intentional?

Reply:
to be in sync with `.go-version`

Reply:
if it's a blocker i can put back the `go-version` action but use it in the `setup-go` composite

Reply:
No strong opinion here, I think this is fine.

Reply:
I would vote for keeping it at the minor version(`1.21`) instead of the patch version(`1.21.3`), unless there is a necessity to stick to that specific patch version. I think that makes it easier to maintain, my 2 cents. 

I believe that's more of a personal preference rather than a right or wrong question, so, it's completely fine to agree to disagree :)

Reply:
@serdardalgic Fair point, I think I agree. Let's use the minor version if it's possible @zekth

I'd also use that in .go-version, if we want them to stay in sync.